PrebidMobile/PrebidMobileCore/src/main/java/org/prebid/mobile/core/Prebid.java

```diff
@@ -236,12 +289,55 @@ private static void handleDFPCustomTargetingUpdate(Object adRequestObj, String a
         if (bundle != null) {
             ArrayList<Pair<String, String>> prebidKeywords = BidManager.getKeywordsForAdUnit(adUnitCode, context);
             if (prebidKeywords != null && !prebidKeywords.isEmpty()) {
-                // retrieve keywords from mopub adview
-                for (Pair<String, String> keywordPair : prebidKeywords) {
-                    bundle.putString(keywordPair.first, keywordPair.second);
-                    usedKeywordKeys.add(keywordPair.first);
+                boolean bidShouldBeLoadedInDemandSdk = false;
+                String bidder = "";
+                for (Pair<String, String> p : prebidKeywords) {
+                    if (PREBID_CREATIVE_LOAD_TYPE.equals(p.first)) {
+                        if (PREBID_DEMAND_SDK.equals(p.second)) {
+                            bidShouldBeLoadedInDemandSdk = true;
+                        }
+                    }
+                    if (PREBID_BIDDER.equals(p.first)) {
+                        bidder = p.second;
+                    }
+                }
+                if (bidShouldBeLoadedInDemandSdk) {
+                    // pass bids that require demand sdk only when rendering is enabled
+                    // save the cache id locally on the ad object for later use
+                    Class mediation_adapter = null;
+                    Set<String> keywords = (Set) callMethodOnObject(adRequestObj, "getKeywords");
+                    for (String keyword : keywords) {
+                        if (PREBID_INTERSTITIAL.equals(keyword)) {
+                            mediation_adapter = getClassFromString(PREBID_DFP_CUSTOM_EVENT_INTERSTITIAL);
+                        } else if (PREBID_BANNER.equals(keyword)) {
+                            mediation_adapter = getClassFromString(PREBID_DFP_CUSTOM_EVENT_BANNER);
+                        }
```

Comment:
Is it possible here to figure out whether or not this is a banner or interstitial based on the adRequestObj? or any other info that doesn't require the publisher to add the additional lines

builder.addCustomEventExtrasBundle(PrebidCustomEventBanner.class, new Bundle());
builder.addKeyword("prebid_banner");

Reply:
I tested that without adding the bundle for PrebidCustomEventBanner.class on the builder, you can't use reflection or other way to have bundle to be passed. Banner and interstitial ad both uses the same ad request object type, so can't figure out banner or interstitial just based on the object.

Reply:
What about based on the adUnitCode? can we pull the adUnit and therefore the type of ad unit?

Reply:
That's a good idea.
But builder.addCustomEventExtrasBundle(PrebidCustomEventBanner.class, new Bundle()); has to be there.